simulator-core/src/test/java/com/evbox/everon/ocpp/functional/provisioning/ResetWithOngoingTransactionTest.java

```diff
@@ -0,0 +1,61 @@
+package com.evbox.everon.ocpp.functional.provisioning;
+
+import com.evbox.everon.ocpp.simulator.message.ActionType;
+import com.evbox.everon.ocpp.simulator.message.Call;
+import com.evbox.everon.ocpp.simulator.station.actions.Plug;
+import com.evbox.everon.ocpp.testutils.ocpp.exchange.Authorize;
+import com.evbox.everon.ocpp.testutils.ocpp.exchange.BootNotification;
+import com.evbox.everon.ocpp.testutils.ocpp.exchange.TransactionEvent;
+import com.evbox.everon.ocpp.testutils.station.StationSimulatorSetUp;
+import com.evbox.everon.ocpp.v20.message.centralserver.ResetRequest;
+import org.junit.jupiter.api.Test;
+
+import java.util.UUID;
+
+import static com.evbox.everon.ocpp.testutils.constants.StationConstants.*;
+import static com.evbox.everon.ocpp.testutils.expect.ExpectedCount.atLeastOnce;
+import static com.evbox.everon.ocpp.testutils.expect.ExpectedCount.times;
+import static com.evbox.everon.ocpp.v20.message.common.IdToken.Type.ISO_14443;
+import static com.evbox.everon.ocpp.v20.message.station.BootNotificationRequest.Reason.REMOTE_RESET;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ResetWithOngoingTransactionTest extends StationSimulatorSetUp {
+
+    @Test
+    void shouldImmediatelyResetWithOngoingTransaction() {
+
+        ocppMockServer
+                .when(BootNotification.request(), times(2))
+                .thenReturn(BootNotification.response());
+
+        ocppMockServer
+                .when(Authorize.request(DEFAULT_TOKEN_ID, ISO_14443))
+                .thenReturn(Authorize.response());
+
+        ocppMockServer
+                .when(BootNotification.request(REMOTE_RESET), atLeastOnce())
+                .thenReturn(BootNotification.response());
+
+        ocppMockServer
+                .when(TransactionEvent.request(), atLeastOnce())
+                .thenReturn(TransactionEvent.response());
+
+        stationSimulatorRunner.run();
+        ocppMockServer.waitUntilConnected();
+
+        triggerUserAction(STATION_ID, new Plug(DEFAULT_EVSE_ID, DEFAULT_CONNECTOR_ID));
+        triggerUserAction(STATION_ID, new com.evbox.everon.ocpp.simulator.station.actions.Authorize(DEFAULT_TOKEN_ID, DEFAULT_EVSE_ID));
```

Comment:
is there another class with Authorize name?

Reply:
Authorize.response and Authorize.request in ocpp/exchange.

And there is a station.actions.Authorize as well.